test/Test/BaseDAO/Token/FA2.hs

```diff
@@ -378,7 +378,7 @@ adminTransferFrozenScenario = uncapsNettest $ do
 
 transferAfterMigrationScenario :: (Monad m) => NettestImpl m -> m ()
 transferAfterMigrationScenario = uncapsNettest $ do
-  ((owner1, op1), (owner2, newAddress1), dao, admin) <- originateBaseDao @()
+  ((owner1, op1), (owner2, newAddress1), dao, admin) <- originateBaseDao @() ()
```

Comment:
This wall of changes where we just add `()` argument passing makes me a bit sad. Maybe let's define `originateTrivialDao` where we can hardcode this `()`? Sounds like a natural next step since we added a notion of trivial DAO.

It won't eliminate the wall of changes (we still need to `s/originateBaseDao/originateTrivialDao` in 100500 places), but should at least make these tests more resistant to possible future changes.

Reply:
What about `originateBaseDaoWithConfig` and `originateBaseDaoWithBalance`? Add their trivial versions as well?

Reply:
Does trivial version of `originateBaseDaoWithConfig` make any sense? "Trivial" means that config is fixed, "WithConfig" means it can be supplied externally :confused: 

Regarding `originateBaseDaoWithBalance` it's not obvious in how many places its trivial version will be useful, but if there are many such places (like 3+), probably it makes sense to add such a helper.

Reply:
> Does trivial version of originateBaseDaoWithConfig make any sense? "Trivial" means that config is fixed, "WithConfig" means it can be supplied externally confused

Well, it could have a different name (`s/trivial/simple`), but such a function still makes sense. In fact, we call `originateBaseDaoWithConfig` with empty contract extra approximately as often as `originateBaseDao`.

> Regarding originateBaseDaoWithBalance it's not obvious in how many places its trivial version will be useful, but if there are many such places (like 3+), probably it makes sense to add such a helper.

There are exactly 3 such places :smile:

Reply:
> Well, it could have a different name (s/trivial/simple), but such a function still makes sense. In fact, we call originateBaseDaoWithConfig with empty contract extra approximately as often as originateBaseDao.

I think "trivial" means 3 things:
1. Empty contract extra.
2. Empty proposal metadata.
3. Trivial config, no special logic.

And usually first two things (contract extra and metadata) are deduced from the config. But not always, there can be a polymorphic non-trivial config. If there are many places where we pass a non-trivial polymorphic config and some constant type parameter(s), we can have a function for that as well.

I see we have a lot of `originateBaseDaoWithConfig @Integer config` in `Test.BaseDAO.Proposal`. Probably it makes sense to have a function defined as `originateBaseDaoWithConfig @Integer config` in that module.

Reply:
AFAIU this discussion is not resolved yet.

Reply:
Yeah, need some time to finish it